dashboard/client/components/+workloads-pods/pod-container-ports.tsx

```diff
@@ -0,0 +1,57 @@
+import "../pod-ports.scss"
+
+import React from "react";
+import { observer } from "mobx-react";
+import { t } from "@lingui/macro";
+import { Pod, IPodContainer } from "../../api/endpoints";
+import { _i18n } from "../../i18n";
+import { apiBase } from "../../api"
+import { observable } from "mobx";
+import { cssNames } from "../../utils";
+import { Notifications } from "../notifications";
+import { Spinner } from "../spinner"
+
+interface Props {
+  pod: Pod;
+  container: IPodContainer;
+}
+
+@observer
+export class PodContainerPorts extends React.Component<Props> {
+  @observable waiting = false;
+
+  async portForward(port: number) {
+    const { pod } = this.props;
+    this.waiting = true;
+    apiBase.post(`/pods/${pod.getNs()}/pod/${pod.getName()}/port-forward/${port}`, {})
+      .catch(error => {
+        Notifications.error(error);
+      })
+      .finally(() => {
+        this.waiting = false;
+      });
```

Comment:
This can be done using a `try { ... } catch { ... } finally { ... }` block and it would be more idiomatic.
```suggestion
    try {
        await apiBase.post(`/pods/${pod.getNs()}/pod/${pod.getName()}/port-forward/${port}`, {});
    } catch (error) {
        Notifications.error(error);
    } finally {
        this.waiting = false;
    }
```

Reply:
What is there is the more modern approach?

Reply:
Not really. The more modern approach would be to use a `try/catch` and `await`.

See: 
https://stackoverflow.com/questions/42453683/how-to-reject-in-async-await-syntax
https://jakearchibald.com/2014/es7-async-functions/
https://itnext.io/error-handling-with-async-await-in-js-26c3f20bc06a